lib/board.rb

```diff
@@ -0,0 +1,41 @@
+require_relative './token'
+
+class Board
+    include Token
+    attr_accessor :size, :positions, :max_positions
+    
+    def initialize(size:)
+        @size = size
+        @max_positions = size * size
+        @positions = (1..max_positions).map {|num| num = empty }
+    end
+
+    def update(position:, token:)
+        positions[position] = token
+        @positions = positions;
+        return positions
+    end
+
+    def is_full?()
+        positions.all? {|position| position != empty }
+    end
+
+    def position_is_available?(position)
+        return positions[position] == empty
+    end
+
+    def empty_positions()
+        positions.each_with_index.reduce([]) do |empty_positions, (position, idx)| 
+            position == empty && empty_positions.push(idx)
```

Comment:
why don't you take advantage of the `is_position_empty` function here? (it's a is a very simple line though so I don't have a strong opinion)